UnitTests/GitUITests/CommandsDialogs/CommitDialog/FormCommitTests.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using CommonTestUtils;
+using GitCommands;
+using GitUI;
+using GitUI.CommandsDialogs;
+using NUnit.Framework;
+using File = System.IO.File;
+using Path = System.IO.Path;
+
+namespace GitUITests.CommandsDialogs.CommitDialog
+{
+    [TestFixture]
+    [Apartment(ApartmentState.STA)]
+    public class FormCommitTests
+    {
+        private static GitModuleTestHelper _moduleTestHelper;
+        private static string _commitHash;
+        private GitUICommands _commands;
+
+        [OneTimeSetUp]
+        public static void OneTimeSetUp()
+        {
+            _moduleTestHelper = new GitModuleTestHelper();
+
+            using (var repository = new LibGit2Sharp.Repository(_moduleTestHelper.Module.WorkingDir))
```

Comment:
💭 In the long run, we would likely see benefits from creating a test state shared across a wider variety of tests. This object can quickly create a repository with a known form, and reset its state between tests so it's easy to write a test that relies on a current commit with particular characteristics. The use of LibGit2Sharp has two advantages:

1. It's a simple and fast API for constructing a repository with a known form
2. It's decoupled from GitExtensions action-tracking features, so it's easy to create a repository during testing without otherwise influencing the GitExtensions behavior